apps/loan/loan-products/src/app/components/loan-products-grid/loan-products-grid.component.ts

```diff
@@ -58,19 +62,23 @@ export class LoanProductsGridComponent implements OnChanges {
   ngOnChanges(changes: SimpleChanges): void {
     // When loan products change, generate action menus
     if (this.loanProducts && changes['loanProducts']) {
-      this.actions = this.actionsGenerate(this.loanProducts);
+      this.actions = this.actionsGenerate(this.loanProducts, this.isLocked);
       this.monthlyPaymentRange.set(
         this.generateMonthlyRange(this.loanProducts)
       );
     }
+    // When the lock status changes
+    if (changes['isLocked']) {
```

Comment:
My only feedback is if you use an `input()` signal then you can use `computed()` for `actions` and I think avoid all this `ngOnChanges` stuff. - https://v17.angular.io/guide/signals#computed-signals

Reply:
Great comment! I had totally forgotten about computed signals. Reactive code is so very much better. Now I just need to do this for the rest of the app.